src/modules/75-ce/components/COInstanceSelector/COInstanceSelector.tsx

```diff
@@ -156,437 +164,457 @@ const COInstanceSelector: React.FC<COInstanceSelectorprops> = props => {
+    additionalProps: { lazy: !isAwsProvider }
   })
   const [selectedRegion, setSelectedRegion] = useState<SelectOption>()
-  const [zonesData, setZonesData] = useState<SelectOption[]>([])
-  const [selectedZone, setSelectedZone] = useState<SelectOption>()
-
-  const { data: resourceGroups, loading: resourceGroupsLoading } = useAllResourceGroups({
-    account_id: accountId, // eslint-disable-line
-    queryParams: {
-      cloud_account_id: gatewayDetails.cloudAccount.id, // eslint-disable-line
-      accountIdentifier: accountId
-    },
-    lazy: !isAzureProvider
-  })
+  const [tagKeys, setTagKeys] = useState<SelectOption[]>([])
+  const [tagValues, setTagValues] = useState<SelectOption[]>([])
+  const [selectedTagPair, setSelectedTagPair] = useState<SelectedTagFilter>()
 
   const {
-    data: zones,
-    loading: zonesLoading,
-    refetch: fetchZones
-  } = useAllZones({
+    data: tagsData,
+    loading: tagsLoading,
+    refetch: fetchTags
+  } = useGetInstancesTags({
     account_id: accountId,
     queryParams: {
       cloud_account_id: gatewayDetails.cloudAccount.id,
       accountIdentifier: accountId,
-      region: ''
+      routingId: accountId,
+      filter: _defaultTo(selectedRegion?.label, '')
     },
     lazy: true
   })
 
   useEffect(() => {
-    setResourceGroupDataFromResponse(resourceGroups?.response)
-  }, [resourceGroups?.response])
-
-  useEffect(() => {
-    if (isAzureProvider) {
-      handleAzureFiltersUpdate()
-    }
-  }, [resourceGroupData])
-
-  useEffect(() => {
-    onResourceGroupSelectCallback(_defaultTo(selectedResourceGroup, null), resourceGroupsLoading)
-  }, [selectedResourceGroup, resourceGroupsLoading])
-
-  useEffect(() => {
-    handleGcpRegionsFilterUpdate()
+    handleRegionsFilterUpdate()
   }, [regionsData])
 
   useEffect(() => {
-    if (selectedRegion) {
-      fetchZones({
-        queryParams: {
-          cloud_account_id: gatewayDetails.cloudAccount.id,
-          accountIdentifier: accountId,
-          region: selectedRegion.label
-        }
-      })
-    }
-    if (isGcpProvider) {
-      onGcpFiltersChangeCallback({ region: selectedRegion }, regionsLoading)
+    if (isAwsProvider) {
+      handleAwsRegionFilterUpdate()
     }
   }, [selectedRegion, regionsLoading])
 
   useEffect(() => {
-    if (zones?.response) {
-      setZonesData(zones.response.map(z => ({ label: z, value: z })))
+    if (tagsData?.response?.length) {
+      const keys = tagsData.response.map(tag => ({ label: tag.key as string, value: tag.key as string }))
+      setTagKeys(keys)
     }
-  }, [zones?.response])
+  }, [tagsData])
 
   useEffect(() => {
-    handleGcpZonesFilterUpdate()
-  }, [zonesData])
+    if (selectedTagPair?.key) {
+      const selectedTag = tagsData?.response?.find(tag => tag.key === selectedTagPair.key)
+      setTagValues(
+        _defaultTo(
+          selectedTag?.values?.map(v => ({ label: v, value: v })),
+          []
+        )
+      )
+    }
+  }, [selectedTagPair?.key])
```

Comment:
Codacy has a fix for the issue: [React Hook useEffect has a missing dependency: 'tagsData?.response'. Either include it or remove the dependency array.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9255099)
```suggestion
  }, [selectedTagPair.key, tagsData?.response])
```